task/fbc-related-image-check/0.1/fbc-related-image-check.yaml

```diff
@@ -30,12 +30,19 @@ spec:
         FAILEDIMAGES=""
         catalog="$(find $(workspaces.workspace.path)/hacbs/fbc-validation/ -name catalog.yaml)"
 
+        # OPM generates catalog file in a way that yaml file could contain json type of data
```

Comment:
I think this comment is a bit misleading. Anyone reading this will immediately think something is off because JSON is a subset of YAML.

The problem here is three fold:
1. `opm` produces a stream of json objects - not an array
2. `catalog.yaml` could also be in yaml format (although I don't fully understand that scenarios that would cause this)
3. `yq` cannot properly handle a stream of objects in a dynamic way

Starting with version 4.27.1, you can force `yq` to read the input as json via the `-p` flag, e.g. `yq -p json $catalog`. However, if the catalog is in YAML format, then this will fail. It would be nice if `yq` could handle this in a more dynamic way. It might be worth filing an issue upstream.

To avoid having to repeat the expression between `yq` and `jq`, as well as, ensuring these tools behave the same way (they don't for more advanced expressions), consider fist converting the file into either json or yaml if needed, then using one of the tools to actually extract values from it, i.e. normalize the input then operate on it instead of trying to operate directly on the different formats.

Reply:
Yes, we can arrange around this in many ways. OR we can recieve actual standardized format and then the solution would be much simpler and easier to work with.
Currently there is two kinds of catalogs formats AFAIK. Yaml file with yaml data, yaml file with json data.
I'm not aware of the way that catalogs are going to be used anyplace else, but this double file/data format will not cause pain only on our side.